nibe/connection/nibegw.py

```diff
@@ -63,24 +76,29 @@ def __init__(
         self._remote_write_port = remote_write_port
 
         self._transport = None
+        self._status = ConnectionStatus.UNKNOWN
 
         self._send_lock = asyncio.Lock()
         self._futures = {}
 
     async def start(self):
         logger.info(f"Starting UDP server on port {self._listening_port}")
 
+        self._set_status(ConnectionStatus.INITIALIZING)
+
         await asyncio.get_event_loop().create_datagram_endpoint(
             lambda: self,
             local_addr=(self._listening_ip, self._listening_port),
             proto=socket.IPPROTO_UDP,
         )
 
     def connection_made(self, transport):
+        self._set_status(ConnectionStatus.LISTENING)
         self._transport = transport
 
     def datagram_received(self, data, addr):
         logger.debug(f"Received {hexlify(data)} from {addr}")
+        self._set_status(ConnectionStatus.CONNECTED)
```

Comment:
One note here. I noticed my chromecast audio sends udp data on port 9999, which is default for nibegw. Its should likely check for the addr here to verify it's from the expected remote address.

Reply:
We did try to add that filter before. But unfortunately when run in docker you do not get real IP address in UDP packets.